dagmc_stats/dagmc_stats.py

```diff
@@ -297,3 +297,157 @@ def get_coarseness(my_core, meshset, entity_ranges, geom_dim):
+    
+    inputs
+    ------
+    my_core : a MOAB Core instance
+    native_ranges : a dictionary containing ranges for each native type in the file (VERTEX, TRIANGLE, ENTITYSET)
+    all_verts : all the vertices that are connected to triangle in the geometry
+    tri_vert_data : numpy structured array that stores the triangle and vertex related data
+    
+    outputs
+    -------
+    gc_all : dictionary in the form of vertex : gaussian curvature value of the vertex
+    """
+    
+    gc_all = {}
+    for vert_i in all_verts:
+        gc_all[vert_i] = gaussian_curvature(vert_i, tri_vert_data)
+    return gc_all
+
+
+def gaussian_curvature(vert_i, tri_vert_data):
+    """
+    Get gaussian curvature value of a vertex
+    Reference: https://www.sciencedirect.com/science/article/pii/S0097849312001203
+    Formula 1
+    
+    inputs
+    ------
+    vert_i : vertex entity handle
+    tri_vert_data : numpy structured array that stores the triangle and vertex related data
+    
+    outputs
+    -------
+    gc : gaussian curvature value of the vertex
+    
+    """
+    
+    vert_entries = tri_vert_data[tri_vert_data['vert']==vert_i]
+    sum_alpha_angles = sum(vert_entries['angle'])
+    gc = np.abs(2*np.pi - sum_alpha_angles)
+    return gc
+
+
+def get_lri(vert_i, gc_all, tri_vert_data, my_core):
+    """
+    Get local roughness value of a vertex
+    Reference: https://www.sciencedirect.com/science/article/pii/S0097849312001203
+    Formula 2, 3
+    
+    inputs
+    ------
+    vert_i : vertex entity handle
+    gc_all : dictionary in the form of vertex : gaussian curvature value of the vertex
+    tri_vert_data : numpy structured array that stores the triangle and vertex related data
+    my_core : a MOAB Core instance
+    
+    outputs
+    -------
+    Lri : local roughness value of the vertex
+    
+    """
+    
+    DIJgc_sum = 0
+    Dii_sum = 0
+    vert_j_list = list(my_core.get_adjacencies(my_core.get_adjacencies(vert_i,2,op_type=0),0,op_type=1))
+    vert_j_list.remove(vert_i)
+    for vert_j in vert_j_list:
+        # get tri_ij_list (the list of the two triangles connected to both vert_i and vert_j)
+        tri_i_list = my_core.get_adjacencies(vert_i, 2, op_type=0)
+        tri_j_list = my_core.get_adjacencies(vert_j, 2, op_type=0)
+        tri_ij_list = list(set(tri_i_list) & set(tri_j_list))
+        # get vertices that are in triangles in tri_ij_list
+        # there will be six elements representing four vertices
+        # vert_i and vert_j will appear twice
+        four_vert_0 = tri_vert_data[tri_vert_data['tri'] == tri_ij_list[0]]
+        four_vert_1 = tri_vert_data[tri_vert_data['tri'] == tri_ij_list[1]]
+        four_vert = np.concatenate([four_vert_0, four_vert_1])
+        
+        #get beta angles
+        beta_angles = []
+        for entry in four_vert:
+            matched = four_vert[four_vert['vert'] == entry['vert']]
+            if len(matched) < 2:
+                beta_angles.append(entry['angle'])
```

Comment:
I know we discussed this logic in person, but can you also add some comments describing what is happening here so others understand.